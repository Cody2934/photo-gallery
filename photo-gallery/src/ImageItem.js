import React, { Component } from 'react'

export default class ImageItem extends Component {
    render() {
        return ( <li key= {this.props.image}>
            {/* this is the information */}
            <div className="info-container">
                <h2>{this.props.image.name}</h2>
                <p className="monster-type">{this.props.image.keyword}</p>
            </div>

            {/* this is the image */}
            <div className="image-container">
                <img
                    alt={this.props.image.url}
                    src={this.props.image.url} />
            </div>
            <p className="description">{this.props.image.description}</p>
        </li>
        )
    }
}