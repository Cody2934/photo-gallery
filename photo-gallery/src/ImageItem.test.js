import React from 'react';
import ImageItem from './ImageItem.js';
import renderer from 'react-test-renderer';
import images from './data.js'

it('renders correctly', () => {
  const tree = renderer
    .create(<ImageItem image={images} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});