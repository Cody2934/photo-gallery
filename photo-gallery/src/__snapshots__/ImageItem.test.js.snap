// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <div
    className="info-container"
  >
    <h2 />
    <p
      className="monster-type"
    />
  </div>
  <div
    className="image-container"
  >
    <img />
  </div>
  <p
    className="description"
  />
</li>
`;
